python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2394,6 +2399,71 @@ def get_handlers(self) -> Sequence[logging.Handler]:
     def store_event(self, event: "EventLogEntry") -> None:
         self._event_storage.store_event(event)
 
+    def should_retry_run(self, run: DagsterRun) -> bool:
```

Comment:
kind of a nit, but does this need to be a method on DagsterInstance? I could imagine it being a separate standalone method that takes in the run and an instance - since its more like a separate, fairly complex piece of business logic that can be computed based on data on the instance

Reply:
no this is a really good point, i moved it to a different file